### PR TITLE
Parse ipv6 addresses

### PR DESCRIPTION
**Pull request by uablrek**
Parse (and use) ipv6 addresses in test programs.

(starting with `sample_test`)

### CODE REVIEW

**Comment by uablrek** [comment]
@lxin Please review when you got time. I have commented out the error return in `apln_test.c` temporary due to #19.

**Comment by lxin** [comment]
@uablrek, thank you for your contributions! I noticed a few areas for improvement, primarily around coding style. Here are my suggestions:

1. Adherence to Linux Kernel Code Style: This repository follows the Linux kernel coding style. Please run checkpatch.pl from the Linux kernel tree against your patch (e.g., 0001-Support-IPv6-in-tests.patch) and address as many warnings and errors as possible. Here's how you can do it:
```
# git clone git://git.kernel.org/pub/scm/linux/kernel/git/netdev/net.git
# cd net/
# ./scripts/checkpatch.pl 0001-Support-IPv6-in-tests.patch
```

2. Variable Declaration Style (Reverse Christmas Tree): When declaring variables, please follow the "reverse Christmas tree" style for alignment and readability. For example:
```
        struct quic_transport_param param = {};
        struct sockaddr_storage ra = {};
        unsigned int param_len, flags;
```
instead of
```
        struct quic_transport_param param = {};
        unsigned int param_len, flags;
        struct sockaddr_storage ra = {};
```

3. Move Declarations to the Start of Functions: To align with Linux kernel style, ensure all variable declarations are at the beginning of a function. For instance:
```
        char const *rc;
        ....
        
        if (argc < 3) {
                printf("%s client <PEER ADDR> <PEER PORT> [ALPN]\n", argv[0]);
                return 0;
        }

        rc = parse_address(argv[2], argv[3], &ra);
```
instead of
```
        if (argc < 3) {
                printf("%s client <PEER ADDR> <PEER PORT> [ALPN]\n", argv[0]);
                return 0;
        }

        char const* rc = parse_address(argv[2], argv[3], &ra);
```

4. Consistency Between printf() and perror(): You've replaced some printf() calls with perror(). Unless you plan to replace all printf() calls for consistency, it’s better to leave the existing printf() statements as they are.

5. Bug Fix: If you don't mind, please also include the fix I shared earlier to ensure proper assignment of host and psk variables:
```
diff --git a/tests/sample_test.c b/tests/sample_test.c
index 365741a..432ba6f 100644
--- a/tests/sample_test.c
+++ b/tests/sample_test.c
@@ -37,8 +37,8 @@ static int do_client(int argc, char *argv[])
                return -1;
        }
 
-       psk  = strcmp(argv[4], "none") ? argv[4] : NULL;
-       host = strcmp(argv[5], "none") ? argv[5] : NULL;
+       host = strcmp(argv[4], "none") ? argv[4] : NULL;
+       psk  = strcmp(argv[5], "none") ? argv[5] : NULL;
```

**Comment by uablrek** [comment]
There is no bug in sample_test.c. The help printout:
https://github.com/lxin/quic/blob/8df04343c4d4bafecb794264ebe3c6a973b50c3a/tests/sample_test.c#L19-L23

says that argv[4] is psk, and argv[5] is hostname. And it is used in alignment with the help printout:
https://github.com/lxin/quic/blob/8df04343c4d4bafecb794264ebe3c6a973b50c3a/tests/sample_test.c#L40-L43

**Comment by uablrek** [comment]
Other issues are addressed:
```
total: 0 errors, 0 warnings, 529 lines checked

(patch-file).patch has no obvious style problems and is ready for submission.
```
`perror` removed

File: tests/alpn_test.c [hunk 1]
```diff
@@ -227,10 +268,10 @@ static int do_server(int argc, char *argv[])
 			printf("socket getsockname error %d\n", errno);
 			return -1;
 		}
-		printf("LOCAL PORT %d\n", ntohs(sa.sin_port));
-		if (preferred_port != ntohs(sa.sin_port)) {
+		printf("LOCAL PORT %d\n", get_port(&sa));
+		if (preferred_port != get_port(&sa)) {
 			printf("preferred port: %d\n", preferred_port);
-			return -1;
+			//return -1;
```
Comment (lxin) [inline tests/alpn_test.c:274]: any reason why this line is commented out?

File: tests/sample_test.c [hunk 2]
```diff
@@ -5,15 +5,34 @@
...
+	hints.ai_family = AF_UNSPEC;
+	hints.ai_socktype = SOCK_DGRAM;
+
+	rc = getaddrinfo(address, port, &hints, &res);
+	if (rc != 0)
+		return gai_strerror(rc);
+	memcpy(sas, res->ai_addr, res->ai_addrlen);
+	freeaddrinfo(res);
+	return NULL;
+}
```
Comment (lxin) [inline tests/sample_test.c:28]: same here, please add a blank line.

**Review by lxin — CHANGES_REQUESTED** [review]
@uablrek This looks much better! thanks.

I have just a few minor comments inline. While addressing them, please update the commit subject to include the "tests: " prefix, for example:

tests: Parse ipv6 addresses

File: tests/alpn_test.c (hunk 1 above)
Comment (uablrek) [reply tests/alpn_test.c:274]: Yes. Please see #19

File: tests/alpn_test.c [hunk 3]
```diff
@@ -5,26 +5,63 @@
...
+		return ntohs(((struct sockaddr_in *)sas)->sin_port);
+	return ntohs(((struct sockaddr_in6 *)sas)->sin6_port);
+}
+static void set_port(struct sockaddr_storage *sas, unsigned short port)
+{
+	if (sas->ss_family == AF_INET)
+		((struct sockaddr_in *)sas)->sin_port = htons(port);
+	else
+		((struct sockaddr_in6 *)sas)->sin6_port = htons(port);
+}
```
[the remark this answers is not in the document]
Comment (uablrek) [reply tests/alpn_test.c:41]: No, sorry I will not do that. Just that sin_port and sin6_port _happens_ to be at the same offset at the moment, it's not right to assume that will always be the case. I am pretty sure that kernel reviewers would agree with me on that.

**Pull request merged**